distribution/tools/server-cli/src/main/java/org/elasticsearch/server/cli/MachineDependentHeap.java

```diff
@@ -8,193 +8,153 @@
 
 package org.elasticsearch.server.cli;
 
-import org.elasticsearch.common.ParsingException;
+import org.elasticsearch.cluster.node.DiscoveryNodeRole;
 import org.elasticsearch.common.settings.Settings;
-import org.elasticsearch.xcontent.XContentParserConfiguration;
-import org.elasticsearch.xcontent.yaml.YamlXContent;
+import org.elasticsearch.node.NodeRoleSettings;
 
 import java.io.IOException;
-import java.io.InputStream;
-import java.nio.file.Files;
-import java.nio.file.Path;
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
 import java.util.Map;
-import java.util.function.Function;
 
 import static java.lang.Math.max;
 import static java.lang.Math.min;
+import static org.elasticsearch.cluster.node.DiscoveryNodeRole.MASTER_ROLE;
+import static org.elasticsearch.cluster.node.DiscoveryNodeRole.ML_ROLE;
+import static org.elasticsearch.cluster.node.DiscoveryNodeRole.REMOTE_CLUSTER_CLIENT_ROLE;
 import static org.elasticsearch.server.cli.JvmOption.isInitialHeapSpecified;
 import static org.elasticsearch.server.cli.JvmOption.isMaxHeapSpecified;
 import static org.elasticsearch.server.cli.JvmOption.isMinHeapSpecified;
 
 /**
  * Determines optimal default heap settings based on available system memory and assigned node roles.
  */
-public final class MachineDependentHeap {
+public class MachineDependentHeap {
     private static final long GB = 1024L * 1024L * 1024L; // 1GB
     private static final long MAX_HEAP_SIZE = GB * 31; // 31GB
     private static final long MIN_HEAP_SIZE = 1024 * 1024 * 128; // 128MB
-    private static final int DEFAULT_HEAP_SIZE_MB = 1024;
-    private static final String ELASTICSEARCH_YML = "elasticsearch.yml";
 
-    private final SystemMemoryInfo systemMemoryInfo;
-
-    public MachineDependentHeap(SystemMemoryInfo systemMemoryInfo) {
-        this.systemMemoryInfo = systemMemoryInfo;
-    }
+    public MachineDependentHeap() {}
 
     /**
      * Calculate heap options.
      *
-     * @param configDir path to config directory
+     * @param nodeSettings the settings for the node
      * @param userDefinedJvmOptions JVM arguments provided by the user
      * @return final heap options, or an empty collection if user provided heap options are to be used
      * @throws IOException if unable to load elasticsearch.yml
      */
-    public List<String> determineHeapSettings(Path configDir, List<String> userDefinedJvmOptions) throws IOException, InterruptedException {
+    public final List<String> determineHeapSettings(
```

Comment:
Does this even need to be an instance method anymore?

Reply:
Yes, because it calls `getHeapSizeMb` which is protected and meant for overriding.